internal/tableland/impl/acl.go

```diff
@@ -7,38 +7,91 @@ import (
 
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/textileio/go-tableland/internal/tableland"
-	"github.com/textileio/go-tableland/pkg/sqlstore"
+	"github.com/textileio/go-tableland/pkg/database"
+	"github.com/textileio/go-tableland/pkg/database/db"
 	"github.com/textileio/go-tableland/pkg/tables"
 )
 
-type acl struct {
-	store sqlstore.SystemStore
+// ACLStore has access to the stored acl information.
+type ACLStore struct {
+	db *database.SQLiteDB
 }
 
 // NewACL creates a new instance of the ACL.
-func NewACL(store sqlstore.SystemStore) tableland.ACL {
-	return &acl{
-		store: store,
+func NewACL(db *database.SQLiteDB) *ACLStore {
+	return &ACLStore{
+		db: db,
 	}
 }
 
+var _ tableland.ACL = (*ACLStore)(nil)
+
 // CheckPrivileges checks if an address can execute a specific operation on a table.
-func (acl *acl) CheckPrivileges(
+func (acl *ACLStore) CheckPrivileges(
 	ctx context.Context,
 	tx *sql.Tx,
+	chainID tableland.ChainID,
 	controller common.Address,
 	id tables.TableID,
 	op tableland.Operation,
 ) (bool, error) {
-	aclRule, err := acl.store.WithTx(tx).GetACLOnTableByController(ctx, id, controller.String())
+	row, err := acl.db.Queries.WithTx(tx).GetAclByTableAndController(ctx, db.GetAclByTableAndControllerParams{
+		ChainID: int64(chainID),
+		TableID: id.ToBigInt().Int64(),
+		UPPER:   controller.Hex(),
+	})
+	if err == sql.ErrNoRows {
+		return false, nil
+	}
+
 	if err != nil {
 		return false, fmt.Errorf("privileges lookup: %s", err)
 	}
 
+	aclRule, err := aclFromSQLtoDTO(row)
+	if err != nil {
+		return false, fmt.Errorf("transforming to dto: %s", err)
+	}
+
 	isAllowed, _ := aclRule.Privileges.CanExecute(op)
 	if !isAllowed {
 		return false, nil
 	}
 
 	return true, nil
 }
+
+func aclFromSQLtoDTO(acl db.SystemAcl) (SystemACL, error) {
```

Comment:
maybe it would be good to add a small docstring here stating what DTO means

Reply:
sounds good